### PR TITLE
Reusable Blocks: Fix Server side rendering

### PR DESCRIPTION
**Pull request by youknowriad**
The server-side rendering of reusable blocks was not working as expected because the attribute "ref" was dropped because it was a number instead of a string.

This PR changes the ref type to "number" which fixes the issue.

**Testing instructions**

 - Create a reusable block
 - Check that it show up properly on the frontend

### CODE REVIEW

**Comment by jaswrks** [comment]
I believe I may have located another problem related to this. PR: https://github.com/WordPress/gutenberg/pull/3882

**Comment by noisysocks** [comment]
Thanks @youknowriad! 

This broke some tests that were expecting `ref` to be a string. I've fixed those. Merging now so that `master` is in a good state 🙂

File: blocks/api/factory.js [hunk 1]
```diff
@@ -142,7 +143,7 @@ export function switchToBlockType( blocks, name ) {
  */
 export function createReusableBlock( type, attributes ) {
 	return {
-		id: uuid(), // Temorary id replaced when the block is saved server side
+		id: +uniqueId(), // Temorary id replaced when the block is saved server side
```
Comment (aduth) [inline blocks/api/factory.js]: Minor: There's nothing in the signature of `uniqueId` that guarantees it returns a numeric value.

Comment (noisysocks) [reply blocks/api/factory.js]: Our tests will pick this up if lodash ever changes their implementation

**Pull request merged**